cli_tools/common/distro/distro.go

```diff
@@ -29,10 +30,42 @@ const (
 	opensuse = "opensuse"
 	rhel     = "rhel"
 	sles     = "sles"
+	slesSAP  = "sles-sap"
 	ubuntu   = "ubuntu"
 	windows  = "windows"
+
+	archX86 = "x86"
+	archX64 = "x64"
+
+	r2 = "r2"
 )
 
+var osFlagExpression = regexp.MustCompile(
+	// Distro is required and is comprised of at least one letter.
+	// There may be two segments, separated by a hyphen.
+	// Examples: ubuntu, sles, sles-sap, windows
+	"^(?P<distro>[a-z]+(?:-[a-z]+)?)" +
+		// Version is required, and is at least one word character.
+		// Examples: 2004, 2008, 2008r2
+		"-(?P<version>[a-z0-9]+)" +
+		// Architecture is optional. The only options are `x86` and `x64`.
+		"(?:-(?P<arch>x86|x64))?" +
+		// License is optional. The only value is `byol`.
+		"(?:-(?P<license>byol))?$")
+
+var architectures = map[string]bool{
+	archX64: true,
+	archX86: true,
+}
+
+// Flags that don't follow `osFlagExpression` and have been
+// replaced with newer versions.
```

Comment:
Minor: state explicitly that the mapping is from legacy to current flag versions